src/backend/window.rs

```diff
@@ -58,19 +63,44 @@ pub struct WindowHandle<'a> {
 	/// The context handle to use.
 	context_handle: ContextHandle<'a>,
 
-	/// The window ID of the managed window.
-	window_id: WindowId,
+	/// The index of the window in [`Context::windows`].
+	index: usize,
 }
 
 impl<'a> WindowHandle<'a> {
 	/// Create a new window handle from a context handle and a window ID.
-	pub fn new(context_handle: ContextHandle<'a>, window_id: WindowId) -> Self {
-		Self { context_handle, window_id }
+	pub fn new(context_handle: ContextHandle<'a>, index: usize) -> Self {
+		Self { context_handle, index }
+	}
+
+	/// Get a reference to the context.
+	fn context(&self) -> &Context {
+		self.context_handle().context
+	}
+
+	/// Get a mutable reference to the context.
+	///
+	/// # Safety
+	/// The current window may not be moved or removed through the returned reference.
+	/// In practise, this means that you may not create or destroy any windows.
+	unsafe fn context_mut(&mut self) -> &mut Context {
+		&mut self.context_handle.context
+	}
+
+	/// Get a reference to the window.
+	fn window(&self) -> &Window {
+		&self.context().windows[self.index]
+	}
+
+	/// Get a mutable reference to the window.
+	fn window_mut(&mut self) -> &mut Window {
+		let index = self.index;
+		unsafe { &mut self.context_mut().windows[index] }
```

Comment:
Although I see the need for mutable reference to the window, couldnt you potentially also destroy the windows as well?

Reply:
No, you can't drop an object through a mutable reference :)